packages/components/Tabs/src/Tabs.tsx

```diff
@@ -0,0 +1,152 @@
+      (ref: React.RefObject<View>) => {
+        setSelectedTabsItemRef(ref);
+      },
+      [setSelectedTabsItemRef],
+    );
+
+    const onChangeTabId = React.useCallback((key: string, index: number) => {
+      if (getTabId) {
+        return getTabId(key, index);
+      }
+      return `${key}-Tab${index}`;
+    }, []);
+
+    // Stores views to be displayed
+    const map = new Map<string, React.ReactNode[]>();
+
+    const state: TabsState = {
+      context: {
+        selectedKey: selectedKey ?? data.selectedKey,
+        onTabsClick: data.onKeySelect,
+        getTabId: onChangeTabId,
+        updateSelectedTabsItemRef: onSelectTabsItemRef,
+        views: map,
+      },
+      info: {
+        headersOnly: headersOnly ?? false,
+        label: !!label,
+      },
+    };
+
+    const styleProps = useStyling(userProps, (override: string) => state[override] || userProps[override]);
+
+    const ariaRoles = {
+      accessibilityRole: 'tablist',
+      accessibilityLabel: ariaLabel || label,
+    };
+
+    const slotProps = mergeSettings<TabsSlotProps>(styleProps, {
+      root: { rest, ref: componentRef, ...ariaRoles },
+      label: { children: label },
+      container: { isCircularNavigation: isCircularNavigation, defaultTabbableElement: selectedTabsItemRef },
+    });
+
+    return { slotProps, state };
+  },
+
+  render: (Slots: ISlots<TabsSlotProps>, renderData: TabsRenderData, ...children: React.ReactNode[]) => {
+    if (renderData.state == undefined) {
+      return null;
+    }
+
+    // Populate the tabsItemKeys array
+    if (children) {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore - TODO, fix typing error
+      renderData.state.context.tabsItemKeys = React.Children.map(children, (child: React.ReactChild) => {
+        if (React.isValidElement(child)) {
+          // Sets default selected tabItem
+          if (renderData.state.context.selectedKey == null && !child.props.disabled) {
+            renderData.state.context.selectedKey = child.props.itemKey;
+          }
+          return child.props.itemKey;
+        }
+      });
+    }
+
+    return (
+      <TabsContext.Provider
+        // Passes in the selected key and a hook function to update the newly selected tabsItem and call the client's onTabsClick callback
+        value={renderData.state.context}
+      >
+        <Slots.root>
+          {renderData.state.info.label && <Slots.label />}
+          <Slots.container>{children}</Slots.container>
+          <Slots.tabPanel>
+            <TabsContext.Consumer>
+              {(context) => !renderData.state.info.headersOnly && context.views.get(context.selectedKey)}
+            </TabsContext.Consumer>
+          </Slots.tabPanel>
+        </Slots.root>
+      </TabsContext.Provider>
+    );
+  },
+
+  settings,
+  slots: {
+    root: View,
+    label: Text,
+    tabPanel: { slotType: View, filter: filterViewProps },
+    container: FocusZone,
```

Comment:
Isn't this also a View?

Reply:
@samuelfreiberg are you referring to the container slot?

Reply:
Container. I was confused because in the .types file, the container is: `ViewProps & FocusZoneProps`

Reply:
The container is a FocusZone because we need its keyboarding properties. We assigned ViewProps in the .types files to be able to style the focusZone (without ts complaining) in order to have row flex around the tabsitems.

Reply:
@samuelfreiberg should FocusZone have ViewProps as a base? Similar to FocusTrapZone?

Reply:
another option might be to do:

```
 <Slots.container>
        <View style={{flexDirection:'row'}}>
              {children}
        </View>
 </Slots.container>
```

but I'm not sure if that's preferable

Reply:
I made a post below, but here's what I said:

I would make a separate slot called stack. Make this a View and have that be a child to the FocusZone. This should contain the styling info for the children of the FocusZone.

If we end up allowing the user to change the orientation (vertical vs. horizontal), and other styling props, it makes more sense to separate this into separate slots.